internal/backend/runtime/omni/controllers/omni/internal/talos/schematic.go

```diff
@@ -43,28 +45,38 @@ func GetSchematicInfo(ctx context.Context, c *client.Client) (SchematicInfo, err
 	}
 
 	var (
-		extensions  []string
-		schematicID string
+		extensions             []string
+		schematicID            string
+		schematicConfiguration schematic.Schematic
 	)
 
-	items.ForEach(func(status *runtime.ExtensionStatus) {
+	err = items.ForEachErr(func(status *runtime.ExtensionStatus) error {
 		name := status.TypedSpec().Metadata.Name
 		if name == constants.SchematicIDExtensionName { // skip the meta extension
 			schematicID = status.TypedSpec().Metadata.Version
 
-			return
+			if status.TypedSpec().Metadata.ExtraInfo != "" {
```

Comment:
Should we also grab the extensions-only ID (the `id` below) from this ExtraInfo? (not sure if it contains it, but if it does, it could be more reliable than computing it ourselves)

Reply:
it doesn't contain it